src/bin/command_prelude.rs

```diff
@@ -240,15 +240,85 @@ pub trait ArgMatchesExt {
     }
 
     fn compile_options<'a>(
+        &self,
+        ws: &Workspace<'a>,
+        mode: CompileMode,
+    ) -> CargoResult<CompileOptions<'a>> {
+        self._compile_options(ws.config(), Some(ws), mode)
+    }
+
+    fn compile_options_for_single_package<'a>(
+        &self,
+        ws: &Workspace<'a>,
+        mode: CompileMode,
+    ) -> CargoResult<CompileOptions<'a>> {
+        let compile_opts = self.compile_options(ws, mode)?;
+        if compile_opts.requested.specs.len() != 1 {
+            ws.current()?;
```

Comment:
I'm personally not a huge fan of doing this sort of verification in argument parsing, could these move into the main library? 

Cargo-as-a-library I think would be easier to use if we didn't shift too much of the validation to cli argument parsing (which may cause panics/weird bugs if this data gets into cargo-as-a-library). Additionally I feel like argument parsing is primarily concerned with sort of syntactically what's passed in rather than interpreting it which happens in later layers. (also helps us keep everything centralized as much as we can)